### PR TITLE
Step 7: Removed SwiftyJSON dependency

### PR DESCRIPTION
**Pull request by divbyzero**
- Removed EvolvRawAllocations.swift
- Added EvolvRawAllocation with tests
- Added EvolvRawAllocationNode with test
- Removed SwiftyJSON from Podfile and podspec
- Updated dependencies and fixed typo in podspec
- Updated EvolvAllocationStore protocol
- Refactored EvolvAllocations class
- Updated TestData in tests

### CODE REVIEW

File: Example/Example/CustomAllocationStore.swift [hunk 1]
```diff
@@ -16,15 +15,15 @@ public class CustomAllocationStore: EvolvAllocationStore {
      implementing the store. You can install SwiftyJSON here: https://cocoapods.org/pods/SwiftyJSON.
```
Comment (phylliswong) [inline Example/Example/CustomAllocationStore.swift]: As SwiftJSON is no longer a dependency, we should update the comments.

File: Tests/ExecutionTest.swift [hunk 2]
```diff
@@ -189,7 +189,7 @@ class ExecutionTest: XCTestCase {
```
Comment (phylliswong) [inline Tests/ExecutionTest.swift:213]: Need a test that demonstrates the execution supports JSON data type. Please add.

File: EvolvKit.podspec [hunk 3]
```diff
@@ -4,7 +4,7 @@ Pod::Spec.new do |s|
   s.version          = '1.0.2'
   s.summary          = 'Autonomous Optimization Tool'
   s.description      = <<-DESC
-  'This SDK is designed to be integrated into projects to allow for autonomous UI optimozations'
+  'This SDK is designed to be integrated into projects to allow for autonomous UI optimizations'
```
Comment (phylliswong) [inline EvolvKit.podspec:6]: 👍

File: Example/Example/CustomAllocationStore.swift (hunk 1 above)
Comment (phylliswong) [reply Example/Example/CustomAllocationStore.swift]: Will also need to update the README where it mentions SwiftyJSON.

File: Source/EvolvAllocationStore.swift [hunk 4]
```diff
@@ -18,7 +18,7 @@ public protocol EvolvAllocationStore {
      - uid: The participant's unique id.
      - Returns: a SwiftyJSON array of allocation if one exists, else an empty SwiftyJSON array.
```
Comment (phylliswong) [inline Source/EvolvAllocationStore.swift]: SwiftyJSON comments

File: Source/EvolvRawAllocationNode.swift [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+public struct EvolvRawAllocationNode {
+    
+    enum NodeType {
+        case unknown
+        case null
+        case number
+        case string
+        case bool
+        case array
+        case dictionary
```
Comment (PhyllisWong) [inline Source/EvolvRawAllocationNode.swift]: is case dictionary to support json data type?

File: Source/EvolvRawAllocationNode.swift [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+        let keyPathKeys = key.components(separatedBy: ".").filter({ $0.isEmpty == false })
+        
+        if keyPathKeys.isEmpty {
+            throw Error.emptyKey
+        }
+        
+        var node = self
+        var keyPath: [String] = []
+        
+        for key in keyPathKeys {
```
Comment (PhyllisWong) [inline Source/EvolvRawAllocationNode.swift]: we will need to be able to drill down to the furthest nested value in a json that could look like = 
{ genome: { ui: { layout: { checkout: { button: { color: "#ffffff", text: "checkout" } , background: { color: "#000000", height: 500 }}}}}. However the json will probably be more complex with more nested values that we will not know in advance. Does this [String: Any] support this?

File: Tests/AllocationsTest.swift [hunk 7]
```diff
@@ -90,13 +85,12 @@ class AllocationsTest: XCTestCase {
     func test_KeyWithEmptyKeyPart() {
```
Comment (PhyllisWong) [inline Tests/AllocationsTest.swift]: lets add a test that demonstrates json as a data type

File: Tests/RawAllocationNodeTest.swift [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+        XCTAssertNotNil(node)
+        XCTAssertEqual(node?.type, .dictionary)
+    }
+    
+    func test_DecodeDictionary() {
+        // given
+        let jsonString1 = "{}"
+        let jsonString2 = "{\"size\": 126}"
+        let jsonString3 = "{\"scale\": 10.2}"
+        let jsonString4 = "{\"isPromo\": true, \"demo\": {\"bar\": 50}}"
```
Comment (phylliswong) [inline Tests/RawAllocationNodeTest.swift:180]: this is great, but can we add more values that are at least 6 levels nested?

File: Tests/RawAllocationTest.swift [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+//  EvolvKit iOS Tests
+//
+//  Created by divbyzero on 15/08/2019.
+//  Copyright © 2019 Evolv. All rights reserved.
+//
+
+import XCTest
+@testable import EvolvKit
+
+class RawAllocationTest: XCTestCase {
```
Comment (phylliswong) [inline Tests/RawAllocationTest.swift:12]: thank you for adding these tests

File: Source/EvolvRawAllocationNode.swift [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+            UInt8.self,
+            UInt16.self,
+            UInt32.self,
+            UInt64.self,
+            Double.self,
+            String.self,
+            [EvolvRawAllocationNode].self,
+            [String: EvolvRawAllocationNode].self
+        ]
+    }
```
Comment (phylliswong) [inline Source/EvolvRawAllocationNode.swift:159]: 👍

**Review by phylliswong — CHANGES_REQUESTED** [review]
I left some feedback in the comments. Please review and let me know if there are any questions.

File: Example/Example/ViewController.swift [hunk 11]
```diff
@@ -7,15 +7,13 @@
 //
 
 import UIKit
-import SwiftyJSON
 import EvolvKit
```
Comment (phylliswong) [inline Example/Example/ViewController.swift:20]: SwiftyJSON in the doc strings

File: Example/Example/CustomAllocationStore.swift (hunk 1 above)
Comment (divbyzero) [reply Example/Example/CustomAllocationStore.swift]: Yep. I'll do it.

File: Tests/ExecutionTest.swift (hunk 2 above)
Comment (divbyzero) [reply Tests/ExecutionTest.swift:213]: Foundation doesn't have JSON struct in Swift. So we need to use Foundation objects.

Dictionary -> [String: Any]
Array -> [Any]
Value -> NSString, NSNumber, NSNull

https://developer.apple.com/documentation/foundation/jsonserialization

I'll add tests for [String: Any] and [Any].

File: Tests/RawAllocationNodeTest.swift (hunk 8 above)
Comment (divbyzero) [reply Tests/RawAllocationNodeTest.swift:180]: I'll add it.

**Review by phylliswong — APPROVED** [review]
these changes look great, thank you for implementing them.

**Pull request merged**